### ISSUE TITLE
Question about using sample packing

### ISSUE THREAD

**Issue by shuangchen2003** [EvolvingLMMs-Lab/LLaVA-OneVision-1.5#54]
I followed the insructions in the `examples/llava_ov_1_5/sample_packing/README.md`, and tried to use the script `examples/llava_ov_1_5/sample_packing/offline_packing_pipeline.sh` to sample packing your 85M datasets in huggingface. 
But I found that the dataset directory structure published in huggingface is different from that your code assumed. 

Precisely, in your code, you assume that `*.paquet` documents are in the second-level. (From code `    dataset = load_dataset(data_path,data_files='*/*.parquet', split="train", streaming=True) ` in `https://github.com/EvolvingLMMs-Lab/LLaVA-OneVision-1.5/blob/e8b6f9c3c1a511b39b82162d7f826237a8df9513/examples/llava_ov_1_5/sample_packing/huggingface_data_parse.py#L80`
But in huggingface, documents `*.paquet` are in the fouth-level like `LLaVA-OneVision-1.5-Mid-Training-85M/imagenet/CN/part00`

I wonder how should I preprocess the huggingface datasets. Should I firstly move out all the paquets? Or should I do sample packing for all sub-datasets one by one ? 

Looking forward to you reply !

**Comment by yiyexy** [comment]
Sure, you can put all the `*.parquet` files into a single folder. We uploaded them to Hugging Face as separate subsets for clarity.

**Comment by yiyexy** [comment]
No, you can do it!

**Comment by yiyexy** [comment]
We train on the entire training set, so just make sure your training data is sufficiently shuffled during training.
